mevislab.github.io/content/tutorials/thirdparty/matplotlib/BaseNetwork.py

```diff
@@ -0,0 +1,96 @@
+firstSlice = None
+startSlice = None
+lastSlice = None
+endSlice = None
+bins = None 
+slices = None
+
+
+def setDefaults():
+  ctx.field("SubImage.fullSize").touch()
+  ctx.field("SubImage.autoApply").value = True
+  ctx.field("Histogram.updateMode").value = "AutoUpdate" 
+  ctx.field("Histogram.xRange").value = "Dynamic Min/Max"
+  ctx.field("Histogram.useZeroAsBinCenter").value = False
+  ctx.field("Histogram.binSize").value = 1.0
+  ctx.field("Histogram.backgroundValue").value = False
+  ctx.field("Histogram.curveType").value = "Area"
+  ctx.field("Histogram.useStepFunction").value = True
+  ctx.field("Histogram.curveStyle").value = 7
+  updateSlices()
+
+def updateSlices():
+  global firstSlice, startSlice, lastSlice, endSlice, bins, slices
+  firstSlice = ctx.field("SubImage.z").value
+  startSlice = int(ctx.field("SubImage.z").value)
+  lastSlice = int(ctx.field("SubImage.sz").value)+1
+  endSlice = int(ctx.field("SubImage.sz").value)
+  bins = ctx.field("Histogram.binSize").value
+  slices = range(startSlice, endSlice)
```

Comment:
These seem to be constants. You may consider using properties instead of global values.